genesyscloud/resource_genesyscloud_telephony_didpool_test.go

```diff
@@ -0,0 +1,115 @@
+package genesyscloud
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/mypurecloud/platform-client-sdk-go/v48/platformclientv2"
+)
+
+func TestAccResourceDidPoolBasic(t *testing.T) {
+	var (
+		didPoolResource1         = "test-didpool1"
+		didPoolStartPhoneNumber1 = "+13175550000"
+		didPoolEndPhoneNumber1   = "+13175550005"
+		didPoolDescription1      = "Test description"
+		didPoolComments1         = "Test comments"
+		didPoolProvider1         = "PURE_CLOUD"
+	)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: providerFactories,
+		Steps: []resource.TestStep{
+			{
+				// Create
+				Config: generateDidPoolResource(
+					didPoolResource1,
+					didPoolStartPhoneNumber1,
+					didPoolEndPhoneNumber1,
+					nullValue, // No description
+					nullValue, // No comments
+					nullValue, // No provider
+				),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("genesyscloud_telephony_did_pool."+didPoolResource1, "start_phone_number", didPoolStartPhoneNumber1),
+					resource.TestCheckResourceAttr("genesyscloud_telephony_did_pool."+didPoolResource1, "end_phone_number", didPoolEndPhoneNumber1),
+					resource.TestCheckResourceAttr("genesyscloud_telephony_did_pool."+didPoolResource1, "description", ""),
+					resource.TestCheckResourceAttr("genesyscloud_telephony_did_pool."+didPoolResource1, "comments", ""),
+					resource.TestCheckResourceAttr("genesyscloud_telephony_did_pool."+didPoolResource1, "pool_provider", ""),
+				),
+			},
+			{
+				// Update
+				Config: generateDidPoolResource(
+					didPoolResource1,
+					didPoolStartPhoneNumber1,
+					didPoolEndPhoneNumber1,
+					strconv.Quote(didPoolDescription1),
+					strconv.Quote(didPoolComments1),
+					strconv.Quote(didPoolProvider1),
+				),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("genesyscloud_telephony_did_pool."+didPoolResource1, "start_phone_number", didPoolStartPhoneNumber1),
+					resource.TestCheckResourceAttr("genesyscloud_telephony_did_pool."+didPoolResource1, "end_phone_number", didPoolEndPhoneNumber1),
+					resource.TestCheckResourceAttr("genesyscloud_telephony_did_pool."+didPoolResource1, "description", didPoolDescription1),
+					resource.TestCheckResourceAttr("genesyscloud_telephony_did_pool."+didPoolResource1, "comments", didPoolComments1),
+					resource.TestCheckResourceAttr("genesyscloud_telephony_did_pool."+didPoolResource1, "pool_provider", didPoolProvider1),
+				),
+			},
+			{
+				// Import/Read
+				ResourceName:      "genesyscloud_telephony_did_pool." + didPoolResource1,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+		CheckDestroy: testVerifyDidPoolsDestroyed,
+	})
+}
+
+func generateDidPoolResource(
+	resourceID string,
```

Comment:
Once you start getting a large number of values being passed into a function you are better off defining a struct that wrappers all these values.  Otherwise it becomes unreadable :)

Reply:
Good point. I created a struct and use a pointer parameter instead.